### PR TITLE
Fix group users comma joins

### PR DESCRIPTION
**Pull request by tbretl**
The block `find_single_assessment_instance` appears in both of these SQL files:
* `pages/studentAssessmentHomework/studentAssessmentHomework.sql`
* `pages/studentAssessmentExam/studentAssessmentExam.sql`

As its name indicates, this block returns an assessment instance, if one exists. Both blocks used comma-joins like this:
```
SELECT
    ai.*
FROM
    assessment_instances AS ai,
    (SELECT *
     FROM group_users AS gi
     JOIN groups AS gr ON gi.group_id = gr.id
     WHERE $user_id = gi.user_id AND gr.deleted_at IS NULL) AS gid
WHERE
    ai.assessment_id = $assessment_id
    AND ai.number = 1
    AND ((ai.group_id = gid.group_id) OR (ai.user_id = $user_id));
```

Unfortunately, these comma-joins cause the block to return nothing (always) if the assessment instance is not associated with group work. So, `studentAssessment*.js` will try to create a new assessment instance, which will fail.

The ultimate consequence of this mistake is that no student will be able to access any assessment instance after the very first time it is created.

This PR eliminates the comma-joins and fixes the bug, restoring access to assessment instances.

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -261,6 +261,7 @@
 
   * Fix default institution in course instance access rules (Tim Bretl).
 
+  * Fix bad comma joins in select_single_assessment_instance block queries (Tim Bretl).
```
Comment (mwest1066) [inline ChangeLog.md:264]: Don't need ChangeLog entries anymore

File: pages/studentAssessmentExam/studentAssessmentExam.sql [hunk 2]
```diff
@@ -2,11 +2,10 @@
 SELECT
     ai.*
 FROM
-    assessment_instances AS ai,
-    (SELECT *
-     FROM group_users AS gi 
-     WHERE $user_id = gi.user_id) AS gid
+    assessment_instances AS ai
+    LEFT JOIN groups AS g ON (g.deleted_at IS NULL)
+    LEFT JOIN group_users AS gu ON (gu.group_id = ai.group_id AND gu.group_id = g.id)
```
Comment (mwest1066) [inline pages/studentAssessmentExam/studentAssessmentExam.sql:7]: I think the following is equivalent and it feels a bit more natural to me:

```suggestion
    assessment_instances AS ai
    LEFT JOIN groups AS g ON (g.id = ai.group_id AND g.deleted_at IS NULL)
    LEFT JOIN group_users AS gu ON (gu.group_id = g.id)
```

File: pages/studentAssessmentHomework/studentAssessmentHomework.sql [hunk 3]
```diff
@@ -2,111 +2,109 @@
...
     ai.*
 FROM
-    assessment_instances AS ai,
-    (SELECT *
-     FROM group_users AS gi
-     JOIN groups AS gr ON gi.group_id = gr.id
-     WHERE $user_id = gi.user_id AND gr.deleted_at IS NULL) AS gid
+    assessment_instances AS ai
+    LEFT JOIN groups AS g ON (g.deleted_at IS NULL)
+    LEFT JOIN group_users AS gu ON (gu.group_id = ai.group_id AND gu.group_id = g.id)
```
Comment (mwest1066) [inline pages/studentAssessmentHomework/studentAssessmentHomework.sql:7]: Same suggestion:

```suggestion
    assessment_instances AS ai
    LEFT JOIN groups AS g ON (g.id = ai.group_id AND g.deleted_at IS NULL)
    LEFT JOIN group_users AS gu ON (gu.group_id = g.id)
```

File: ChangeLog.md (hunk 1 above)
Comment (tbretl) [reply ChangeLog.md:264]: haha, missed the memo 😄 - reverted

File: pages/studentAssessmentExam/studentAssessmentExam.sql (hunk 2 above)
Comment (tbretl) [reply pages/studentAssessmentExam/studentAssessmentExam.sql:7]: agreed, done, thanks!

File: pages/studentAssessmentHomework/studentAssessmentHomework.sql (hunk 3 above)
Comment (tbretl) [reply pages/studentAssessmentHomework/studentAssessmentHomework.sql:7]: agreed, done, thanks!

**Pull request merged**